### PR TITLE
Changes requested for DR Factory:

### PR DESCRIPTION
**Pull request by sebwolf-de**
* Remove strange comment
* Rename Products -> DynamicRuptureTuple
* Reference -> shared_ptr
* use default and override

### CODE REVIEW

File: src/DynamicRupture/Parameters.h [hunk 1]
```diff
@@ -43,67 +43,67 @@ struct DRParameters {
...
 
     // filename of the yaml file describing the fault parameters
-    initializers::updateIfExists(yamlDrParams, "modelfilename", drParameters.faultFileName);
+    initializers::updateIfExists(yamlDrParams, "modelfilename", drParameters->faultFileName);
   }
 
-  const YAML::Node& yamlElementwiseParams = params["elementwise"];
-  if (params["elementwise"]) {
+  const YAML::Node& yamlElementwiseParams = params->operator[]("elementwise");
+  if (params->operator[]("elementwise")) {
```
Comment (ravil-mobile) [inline src/DynamicRupture/Parameters.h:90]: Maybe `*param["elementwise"]` or `(*param)["elementwise"]`

File: src/DynamicRupture/Parameters.h [hunk 2]
```diff
@@ -43,67 +43,67 @@ struct DRParameters {
...
     initializers::convertStringToMask(yamlElementwiseParams["outputmask"].as<std::string>(), mask);
-    drParameters.isRfOutputOn = drParameters.isRfOutputOn || mask[9];
-    drParameters.isDsOutputOn = drParameters.isDsOutputOn || mask[10];
+    drParameters->isRfOutputOn = drParameters->isRfOutputOn || mask[9];
+    drParameters->isDsOutputOn = drParameters->isDsOutputOn || mask[10];
   }
-  const YAML::Node& yamlPickpointParams = params["pickpoint"];
-  if (params["pickpoint"]) {
+  const YAML::Node& yamlPickpointParams = params->operator[]("pickpoint");
+  if (params->operator[]("pickpoint")) {
```
Comment (ravil-mobile) [inline src/DynamicRupture/Parameters.h:98]: the same as above

**Comment by ravil-mobile** [comment]
let me try this branch with GPU offloading

**Comment by ravil-mobile** [comment]
Everything works with GPUs

File: src/DynamicRupture/Parameters.h (hunk 1 above)
Comment (krenzland) [reply src/DynamicRupture/Parameters.h:90]: agreed

**Review by krenzland — CHANGES_REQUESTED** [review]
Looks good to me, calling operator[] is a bit weird though. Any reason?

**Review by krenzland — APPROVED** [review]

**Pull request merged**